### PR TITLE
Added Mentorship Guidelines

### LINKED ISSUE #3471: Create Mentorship Guidelines document
[Help wanted in editting the issue]

### Please describe the problem (or idea)
Create Mentorship Guidelines document at docs/Programs/Mentorship. We will give link to this document to all the potential NEW mentors so that they will be able to understand our system in 5-10 minutes.

> What happened just before the problem occurred? Or what problem could this idea solve?

The crowd at public lab is increasing. The number of mentors this year is quite large than we had previous years. There is a need to properly document the things which mentors needs to do while working at Public lab group.

> What did you expect to see ?

Documentation should clearly list the flow of events at public labs. That is people need to do single first timer only issues. Then they need to do help wanted issues. Mentors have to assist them in these fields. Explain what first timer and help wanted issues means. Then mentors need to guide students to create first timer issues and help in reviewing prs. We are moving towards a self suistaining system approach where people makes pr and people review the pr amongst themselves

Documentation should clearly list about mentorship programs currenlty pl takes part in. With their links

 if you are creating a gci candidate issue format it properly with
lines like
We are preparing to participate in Google Code-in, and have reserved this issue for participants in GCI - but we'd love to have your help with another one! Please check out https://code.publiclab.org to see more.
If it is a first timer issue than use first timer template
also provide code links
What people are doing from a week at PL is they create issues without proper formatting
In issues for gci we need to list the potential solutions also.
If you are creating an issue for gci you need to put them in the staging list for gci also
#3276
For gci issue you need to create the same task on the gci dashboard
You need to write the instance count
ie how many people can take that
You need to tell the category of the issue in the issue description
and things like that

****

## Thank you!

Your help makes Public Lab better! We *deeply* appreciate your helping refine and improve this site. 

To learn how to write really great issues, which increases the chances they'll be resolved, see:

https://publiclab.org/wiki/developers#Contributing+for+non-coders

### PR DESCRIPTION
**Pull request by rbhatia46**
Fixes #3471 
Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [x] code is in uniquely-named feature branch and has no merge conflicts
* [x] PR is descriptively titled
* [x] PR body includes `fixes #0000`-style reference to original issue #
* [x] ask `@publiclab/reviewers` for help, in a comment below

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software

We have a loose schedule of reviewing and pulling in changes every Tuesday and Friday, and publishing changes on Fridays. 

Thanks!

### CODE REVIEW

**Comment by rbhatia46** [comment]
I tried accumulating some points while preparing this document. Please review and suggest for changes, if any @jywarren @SidharthBansal. 
Thanks!

**Comment by Souravirus** [comment]
Nice work @rbhatia46

**Comment by SidharthBansal** [comment]
Well done.

**Comment by SidharthBansal** [comment]
Please tell that we can open issues at any repository belonging to public lab whether it is on ruby on rails or node js or js. All have link to all repositories with their tech stack. Mainly public lab editor, pl website repo ie plots2, image sequencer, leaflet, plotsbot etc

Many mentors are asking me can we open issue in ..... Repo. So this explanation will help us there.

File: doc/MENTOR_GUIDELINES.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+* When creating an issue for GCI, you need to list the potential solutions too.
+
+* A GCI issue needs to be put in the [staging list](https://github.com/publiclab/plots2/issues/3276) as well.
+
+* You also need to mention the category of the issue(Code, Design, Documentation/training, Quality Assurance, Outreach) and tell the instance count, i.e., how many people can take that issue in the issue description.
+
+* For GCI issue you need to create the same task on the GCI dashboard as well.
+
+* You need to use extra-friendly tone at Public Lab even if people are not able to commit to send a PR on time.
```
Comment (SidharthBansal) [inline doc/MENTOR_GUIDELINES.md]: Can you please change this line? I was just explaining this to you. We can't write that people will not be able to complete issues in the documentation

File: doc/MENTOR_GUIDELINES.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
+### How to be a GCI mentor
+
+To be a GCI mentor at Public labs, there are some prerequisites that we expect you to follow before we can invite you to be a mentor :
+
+1. Solve a [first-timers issue](https://code.publiclab.org/). Provide the link to the first timer issue's pull request (PR) once it's merged.
+
+2. Solve a [help-wanted issue](https://github.com/publiclab/plots2/labels/help-wanted). Help wanted issues are issues which are not labelled as `gci-candidate` neither as `fto-candidate` nor as `first-timer-only`. Provide link to such issues' merged PR.
```
Comment (SidharthBansal) [inline doc/MENTOR_GUIDELINES.md]: It is 'fto' label. Please correct it.

**Comment by rbhatia46** [comment]
Thanks, @sagarpreet-chadha @Souravirus. 
@SidharthBansal  I have made the changes you have suggested. If there are still some changes needed, please let know. 
Thanks!

**Comment by SidharthBansal** [comment]
Also add PR review guidelines
1. PR should use 'Fixes #issue_no" in PR body
2. Code should be modular.
3. Tests must be written if needed. All tests must pass before approval.
4. Suggest code changes if any. Else approve the PR by commentting in the PR.



Also have link to more resources ie publiclab.org/gsoc


@jywarren I read the pl.org/Gsoc doc but it was Gsoc specific. Hence we both are creating this. Hope you will like this. This will reduce our work on gitter chat to guide people

**Comment by jywarren** [comment]
Hi! Shall we begin by merging this and then making some additional `first-timers-only` issues from remaining additions?

**Comment by jywarren** [comment]
And yes, i do like it a lot!!!! Thanks!!!

**Comment by jywarren** [comment]
I copied in the remaining additions to the original issue for others to take on as their FTO issue. Merging this one!

**Comment by jywarren** [comment]
Thanks, everyone! 👍 🎉

**Pull request merged**